src/utilities/fastify/server.ts

```diff
@@ -40,9 +41,21 @@ export async function makeFastify(appPlugin: FastifyPluginAsync, customLogger?:
 
   await server.register(fastifyMongoose);
 
+  const internalServerError = 'Internal server error';
+  server.setErrorHandler(async (error, _request, reply) => {
+    if (error.statusCode !== undefined && error.statusCode < HTTP_STATUS_CODES.INTERNAL_ERROR) {
+      await reply.send(error);
+      return;
+    }
```

Comment:
Had to implement this check. 
This hook catches all the errors including 4xx

Reply:
Makes sense, although it's a shame Fastify doesn't make it easier.